src/v/redpanda/application.cc

```diff
@@ -1764,6 +1764,35 @@ void application::start_runtime_services(
                 }
             });
       });
+    ssx::background = feature_table.invoke_on_all(
+      [this](features::feature_table& ft) {
+          return ft.await_feature(features::feature::rpc_transport_unknown_errc)
+            .then([this] {
+                if (ss::this_shard_id() == 0) {
+                    vlog(
+                      _log.info, "All nodes support unknown RPC error codes");
+                }
+                // Older versions of Redpanda don't properly parse error codes
+                // they don't know about.
+                _rpc.local().set_use_service_unavailable();
+            })
+            .handle_exception([this](const std::exception_ptr& e) {
+                try {
+                    std::rethrow_exception(e);
+                } catch (ss::abort_requested_exception&) {
+                    // Shutting down
+                } catch (...) {
+                    // Should never happen, abort is the only exception that
```

Comment:
This comment is giving me deja-vu: is this copy-paste?  Maybe a higher level await_feature method like `await_feature_then` could wrap this error handling pattern.

Reply:
Good idea, this seems like it'd be generally useful to activate usage of features.